src/frontend/components/ViewersList/InstructorViewersList/index.tsx

```diff
@@ -0,0 +1,170 @@
+import React from 'react';
+import { Box, Button, List } from 'grommet';
+import { AddCircle } from 'grommet-icons';
+import { defineMessages, useIntl } from 'react-intl';
+
+import {
+  useParticipantsStore,
+  ParticipantType,
+} from 'data/stores/useParticipantsStore';
+import { Video } from 'types/tracks';
+import { ViewersListHeader } from 'components/ViewersList/components/ViewersListHeader';
+import { ViewersListItem } from 'components/ViewersList/components/ViewersListItem';
+import { converse } from 'utils/window';
+import { ViewersListItemContainer } from 'components/ViewersList/components/ViewersListItemContainer';
+import { ViewersListTextButton } from 'components/ViewersList/components/ViewersListTextButton';
+
+const messages = defineMessages({
+  demands: {
+    defaultMessage: 'Demands',
+    description:
+      'Participants asking for going on stage are displayed under this label.',
+    id: 'components.ViewersList.demands',
+  },
+  onStage: {
+    defaultMessage: 'On stage',
+    description: 'On-stage participants are displayed under this label.',
+    id: 'components.ViewersList.onStage',
+  },
+  otherViewers: {
+    defaultMessage: 'Other participants',
+    description: 'Connected participants are displayed under this label.',
+    id: 'components.ViewersList.otherViewers',
+  },
+  acceptButton: {
+    defaultMessage: 'Accept',
+    description:
+      'The text displayed in the button in charge of accepting on-stage request, in the viewers list.',
+    id: 'components.ViewersList.acceptButton',
+  },
+  endOnStageButton: {
+    defaultMessage: 'Terminate',
+    description:
+      'The text displayed in the button in charge of making students on stage exiting the stage, in the viewers list.',
+    id: 'components.ViewersList.endOnStageButton',
+  },
+});
+
+interface InstructorViewersListProps {
+  video: Video;
+}
+
+export const InstructorViewersList = ({
+  video,
+}: InstructorViewersListProps) => {
+  const participants = useParticipantsStore((state) => state.participants);
```

Comment:
eventually consider memoize all computation

Reply:
The participant store will not change with the websocket.

Reply:
done here : [5162c0b](https://github.com/openfun/marsha/pull/1443/commits/5162c0bb1a630529314df6b79deb02c50dbf7c2f)